kornia/feature/laf.py

```diff
@@ -0,0 +1,381 @@
+        - Input: :math:`(B, N, 5)`
+        - Output:  :math:`(B, N, 2, 3)`
+    Example:
+        >>> input = torch.ones(1, 10, 5)  # BxNx5
+        >>> output = kornia.ell2LAF(input)  #  BxNx2x3
+    """
+    n_dims = len(ells.size())
+    if (n_dims != 3):
+        raise TypeError(
+            "ellipse shape should be must be [BxNx5]. "
+            "Got {}".format(ells.size()))
+    B, N, dim = ells.size()
+    if (dim != 5):
+        raise TypeError(
+            "ellipse shape should be must be [BxNx5]. "
+            "Got {}".format(ells.size()))
+    ell_shape = torch.cat([torch.cat([ells[:, :, 2:3], ells[:, :, 3:4]], dim=2).unsqueeze(2),
+                           torch.cat([ells[:, :, 3:4], ells[:, :, 4:5]], dim=2).unsqueeze(2)],
+                           dim=2).view(-1, 2, 2)
+    out = torch.matrix_power(torch.cholesky(ell_shape, False), -1).view(B, N, 2, 2)
+    out = torch.cat([out, ells[:, :, :2].view(B, N, 2, 1)], dim=3)
+    return out
+
+
+def LAF2pts(LAF: torch.Tensor, n_pts: int = 50) -> torch.Tensor:
+    """
+    Converts LAFs to boundary points of the regions + center.
+    Used for local features visualization, see visualize_LAF function
+    Args:
+        LAF: (torch.Tensor).
+        n_pts: number of points to output
+
+    Returns:
+        pts: (torch.Tensor) tensor of boundary points
+
+    Shape:
+        - Input: :math:`(B, N, 2, 3)`
+        - Output:  :math:`(B, N, n_pts, 2)`
+    """
+    n_dims = len(LAF.size())
+    if (n_dims != 4):
+        raise TypeError(
+            "LAF shape should be must be [Nx2x3] or [BxNx2x3]. "
+            "Got {}".format(LAF.size())
+        )
+    B, N, _, _ = LAF.size()
+    pts = torch.cat([torch.sin(torch.linspace(0, 2 * math.pi, n_pts - 1)).unsqueeze(-1),
+                     torch.cos(torch.linspace(0, 2 * math.pi, n_pts - 1)).unsqueeze(-1),
+                     torch.ones(n_pts - 1, 1)], dim=1)
+    # Add origin to draw also the orientation
+    pts = torch.cat([torch.tensor([0, 0, 1.]).view(1, 3), 
+                     pts],
+                    dim=0).unsqueeze(0).expand(B * N, n_pts, 3)
+    pts = pts.to(LAF.device).to(LAF.dtype)
+    aux = torch.tensor([0, 0, 1.]).view(1, 1, 3).expand(B * N, 1, 3)
+    HLAF = torch.cat([LAF.view(-1, 2, 3), aux.to(LAF.device).to(LAF.dtype)], dim=1)
+    pts_h = torch.bmm(HLAF, pts.permute(0, 2, 1)).permute(0, 2, 1)
+    return kornia.convert_points_from_homogeneous(
+           pts_h.view(B, N, n_pts, 3))
+
+
+def get_LAF_pts_to_draw(img: torch.Tensor,  # pragma: no cover
+                  LAF: torch.Tensor,
+                  img_idx: int = 0):
+    """
+    Returns numpy array for drawing LAFs
+    """
+    pts = LAF2pts(LAF[img_idx:img_idx + 1])[0]
+    pts_np = pts.detach().permute(1, 0, 2).cpu().numpy()
+    return (pts_np[:, :, 0], pts_np[:, :, 1])
+
+
+def visualize_LAF(img: torch.Tensor,  # pragma: no cover
+                  LAF: torch.Tensor,
+                  img_idx: int = 0,
+                  color = 'r'):
```

Comment:
if this it's only the function that uses matplotlib, I really suggest to remove it. Since we provide `get_LAF_pts_to_draw(...)` it can be well documented saying that the intention is to be used with matplotlib or other visualization libs. @lferraz has some opinions about that :D

Reply:
I still consider this function as a killer feature :P 
And try-except import of the matplotlib solves issue, imo

Reply:
I have the feeling that adding a visualisation system is out of the scope of this PR. This should be decided before. E.g. why matplotlib? 
I consider if a visualisation library is used it should be used across all the library not only in one very specific place. I am not against the visualisation, I love it, but I do not understand why here and not in other places.